File: src/main/java/com/tom/Hello.java
package com.tom;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello world");
//        new Person().這個類別我們稱它為物件CTRL+ALT+空白鍵=這裡面有小括號的圖示它叫做功能也叫做方法
//        new Person().hello();
//        or
        //先定義一個"專門"存放Person物件的物件
        //它的名稱叫做p 等於new Person
        //這樣的一個語法就是在等號右邊先生出一個Person的物件
        //再將這個Person的物件地址寫在一張叫p的紙上面
        //所以我們可以簡稱這個p就是代表了記憶體裡面的Person物件
        //所以我們也可以在接下來的程式碼在p這個物件去呼叫他身上的Hello方法取得結果
        Person p = new Person();
        p.hello();
        /**Person是物件
         * 把它儲存起來
         * 再利用物件去呼叫hello這個方法*/
    }
}

File: src/main/kotlin/com/tom/Hello.kt
package com.tom

////定義類別名稱
//
//fun main() {
////    print("Hello kotlin")
//// 這樣就已經是一個物件了在這邊我們可以直接呼叫Hello
////    Human().hello()
//}
//
////在Kotlin這邊寫類別只要在這邊加上class {}類別名稱再加上大括號就可以了
//class Human {
//    //到這個時候可以發現Person的JAVA專案應該已經衝突了
//    fun hello() {
//        print("Hello kotlin")
//    }
//}
//-----------------------------------------------------------------------
//定義類別名稱

fun main() {
//儲存一個物件
    //可以使用val或是來定義var
    //val(value)代表當我定義完這個物件後就不能再改變他的值
    val h = Human()
    h.hello()
}

//在Kotlin這邊寫類別只要在這邊加上class {}類別名稱再加上大括號就可以了
class Human {
    //到這個時候可以發現Person的JAVA專案應該已經衝突了
    fun hello() {
        print("Hello kotlin")
    }
}